### PR TITLE
Added Caddyfile for a GNU Social installation.

### PR DESCRIPTION
**Pull request by stormdragon2976**
This is a working Caddyfile for GNU Social. It includes README.md with basic info.

### CODE REVIEW

File: gnusocial/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## GNU Social
+
+This Caddyfile should work with any branch of [GNU Social][gs]. For prerequisites and installation information, see the included README file.
+
+To get the latest version, enter the following where you want your [gs][GNU Social] installation located:
+    git clone https://git.gnu.io/gnu/gnu-social.git
+If you are feeling adventurous, you can obtain the bleeding edge version with:
+    git checkout nightly
+while in your gnusocial directory.
```
Comment (mholt) [inline gnusocial/README.md]: If you could fix the code block formatting (need empty lines on both sides) that would be good :)

File: gnusocial/Caddyfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+example.com www.example.com {
+
+# Your fastcgi for php-fpm will be different if you are listening on a socket
+# or port. Here Are examples for both methods.
+# fastcgi / 127.0.0.1:9000 php
+# fastcgi / /var/run/php-fpm/php-fpm.sock php
+gzip
```
Comment (mholt) [inline gnusocial/Caddyfile]: Is gzip needed for this app?

File: gnusocial/Caddyfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+tls <your email address>
```
Comment (mholt) [inline gnusocial/Caddyfile]: I don't recommend using the `tls` directive; let's leave it out of this example

File: gnusocial/Caddyfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+rewrite {
+    to {path} {path}/ /index.php?p={path}
+    }
```
Comment (mholt) [inline gnusocial/Caddyfile]: Move the closing `}` to the beginning of the line

File: gnusocial/Caddyfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
+example.com www.example.com {
```
Comment (mholt) [inline gnusocial/Caddyfile]: Remove the curly brace at the end of this line

File: gnusocial/Caddyfile [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+# Your fastcgi for php-fpm will be different if you are listening on a socket
+# or port. Here Are examples for both methods.
```
Comment (mholt) [inline gnusocial/Caddyfile]: "Uncomment one" :wink:

File: gnusocial/Caddyfile [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+# fastcgi / /var/run/php-fpm/php-fpm.sock php
+gzip
+tls <your email address>
+
+rewrite {
+    to {path} {path}/ /index.php?p={path}
+    }
+
+root <Path to your installation, e.g. /var/www/html/gnusocial>
+}
```
Comment (mholt) [inline gnusocial/Caddyfile]: Remove this closing brace

File: gnusocial/Caddyfile [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+# fastcgi / 127.0.0.1:9000 php
+# fastcgi / /var/run/php-fpm/php-fpm.sock php
+gzip
+tls <your email address>
+
+rewrite {
+    to {path} {path}/ /index.php?p={path}
+    }
+
+root <Path to your installation, e.g. /var/www/html/gnusocial>
```
Comment (mholt) [inline gnusocial/Caddyfile]: Just do `root /var/www/html/gnusocial` -- they'll figure it out. More real this way.

**Comment by stormdragon2976** [comment]
I pushed the requested changes. I think I got them all.

**Pull request merged**